python/paddle/fluid/tests/unittests/test_multimarginloss.py

```diff
@@ -0,0 +1,348 @@
+        label = paddle.to_tensor([0], dtype='int32')
+        self.assertRaises(ValueError,
+                          paddle.nn.functional.multi_margin_loss,
+                          input=input,
+                          label=label,
+                          reduction="unsupport reduction")
+        paddle.enable_static()
+
+    def test_MultiMarginLoss_dimension(self):
+        paddle.disable_static()
+
+        input = paddle.to_tensor([[0.1, 0.3], [1, 2]], dtype='float32')
+        label = paddle.to_tensor([0, 1, 1], dtype='int32')
+
+        self.assertRaises(
+            ValueError,
+            paddle.nn.functional.multi_margin_loss,
+            input=input,
+            label=label,
+        )
+        MMLoss = paddle.nn.MultiMarginLoss()
+        self.assertRaises(
+            ValueError,
+            MMLoss,
+            input=input,
+            label=label,
+        )
+        paddle.enable_static()
+
+    def test_MultiMarginLoss_p(self):
+        p = 2
+        shape = (2, 2)
+        reduction = 'mean'
+        place = paddle.CPUPlace()
+        input = np.random.uniform(0.1, 0.8, size=shape).astype(np.float64)
+        label = np.random.uniform(0, input.shape[1],
+                                  size=(2, )).astype(np.int64)
+        expected = calc_multi_margin_loss(input=input,
+                                          p=p,
+                                          label=label,
+                                          reduction=reduction)
+
+        dy_result = test_dygraph(
+            place=place,
+            p=p,
+            input=input,
+            label=label,
+            reduction=reduction,
+        )
+
+        static_result = test_static(
+            place=place,
+            p=p,
+            input_np=input,
+            label_np=label,
+            reduction=reduction,
+        )
+        self.assertTrue(np.allclose(static_result, expected))
+        self.assertTrue(np.allclose(static_result, dy_result))
+        self.assertTrue(np.allclose(dy_result, expected))
+        static_functional = test_static(place=place,
+                                        p=p,
+                                        input_np=input,
+                                        label_np=label,
+                                        reduction=reduction,
+                                        functional=True)
+        dy_functional = test_dygraph(place=place,
+                                     p=p,
+                                     input=input,
+                                     label=label,
+                                     reduction=reduction,
+                                     functional=True)
+        self.assertTrue(np.allclose(static_functional, expected))
+        self.assertTrue(np.allclose(static_functional, dy_functional))
+        self.assertTrue(np.allclose(dy_functional, expected))
+
+    def test_MultiMarginLoss_weight(self):
+        shape = (2, 2)
```

Comment:
I suggest using different size for batch_size and number of classes to ensure that weight's is (C,) instead of `(N, )`.

Reply:
Done, use different size for N and C